shared/utility/math/quaternion.hpp

```diff
@@ -35,13 +35,19 @@ namespace utility::math::quaternion {
               typename QType  = std::remove_cv_t<std::remove_reference_t<decltype(*std::declval<Iterator>())>>,
               typename Scalar = typename QType::Scalar>
     inline QType mean(const Iterator& begin, const Iterator& end) {
+        if (std::distance(begin, end) == 0) {
+            return QType::Identity();
+        }
         // Initialise our accumulator matrix
         Eigen::Matrix<Scalar, 4, 4> A = Eigen::Matrix<Scalar, 4, 4>::Zero();
 
         // Accumlate the quaternions across all particles
         for (Iterator it = begin; it != end; ++it) {
             // Convert quaternion to vec4
             Eigen::Matrix<Scalar, 4, 1> q(it->coeffs());
+            if (q.w() < Scalar(0)) {
+                q *= Scalar(-1);
```

Comment:
If you can do it like this, you should do it like this in the other spot too. it's cleaner (but I thought you couldn't do it like this?)

Reply:
You can for a vector, we could use `q.coeffs() *= Scalar(-1);`

Reply:
That seems good to me.